libs/features/mas/web-components/src/variants/variant-layout.js

```diff
@@ -0,0 +1,103 @@
+import { html } from 'lit';
+
+export class VariantLayout {
+  static styleMap = {};
+
+  card;
+
+  insertVariantStyle() {
+    if (!VariantLayout.styleMap[this.card.variant]) {
```

Comment:
nit: with above suggestion, it could be written as `if (!styleMap[this.card.variant]) {`

Reply:
since we append the style below, I would call the method `appendVariantStyles`